src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerSearchService.kt

```diff
@@ -64,4 +65,52 @@ class PrisonerSearchService(
     }
     return prisoners
   }
+
+  /**
+   * Get a prisoner info by prisonerNumber
+   * Requires role PRISONER_SEARCH or VIEW_PRISONER_DATA or PRISONER_SEARCH__PRISONER__RO
+   */
+  suspend fun getPrisonerInfo(prisonerNumber: String): Prisoner {
+    return prisonerSearchWebClient.get()
+      .uri(
+        "/prisoner/{prisonerNumber}?" +
+          "responseFields={responseFields}",
+        mapOf(
+          "prisonerNumber" to prisonerNumber,
+          "responseFields" to responseFields,
+        ),
+      )
+      .retrieve()
+      .awaitBody<Prisoner>()
+  }
+
+  /**
+   * Get a prisoner info by bookingId
+   * Requires role GLOBAL_SEARCH or PRISONER_SEARCH
+   */
+  suspend fun getPrisonerInfo(bookingId: Long): Prisoner {
+    val results = prisonerSearchWebClient.post()
+      .uri(
+        "/prisoner-search/booking-ids?" +
```

Comment:
NOTE: There isn't a way to get a single prisoner by bookingId but there is an endpoint to get multiple, so using that but to get information by passing a single `bookingId` in the request body.

Reply:
We need to check that it can handle booking Id - Not sure let me check because search only hold the LATEST booking ID not older ones.  I need to check WHEN it looks up by bookingID 

If not we may still have to use prisonId for this endpoint